numbat/src/command.rs

```diff
@@ -177,112 +479,79 @@ impl<'a> CommandParser<'a> {
-            }
-            CommandKind::List => {
-                let items = self.inner.args.next();
-
-                if self.inner.args.next().is_some() {
-                    return Err(self.err_through_end_from(2, "`list` takes at most one argument"));
-                }
+#[cfg(test)]
+mod test {
+    use super::*;
 
-                let items = match items {
-                    None => None,
-                    Some("functions") => Some(ListItems::Functions),
-                    Some("dimensions") => Some(ListItems::Dimensions),
-                    Some("variables") => Some(ListItems::Variables),
-                    Some("units") => Some(ListItems::Units),
-                    _ => {
-                        return Err(self.err_at_idx(
-                            1,
-                            "if provided, the argument to `list` must be \
-                            one of: functions, dimensions, variables, units",
-                        ));
-                    }
-                };
+    #[derive(Debug, Clone, PartialEq)]
+    pub enum BareCommand<'a> {
+        Help,
+        Info { item: &'a str },
+        List { items: Option<ListItems> },
+        Clear,
+        Save { dst: &'a str },
+        Quit,
+    }
 
-                Command::List { items }
+    impl<'b, Editor> Command<'_, 'b, Editor> {
+        fn into_bare(self) -> BareCommand<'b> {
+            match self {
+                Command::Help { print_fn: _ } => BareCommand::Help,
+                Command::Info { item, .. } => BareCommand::Info { item },
+                Command::List { items, .. } => BareCommand::List { items },
+                Command::Clear { clear_fn: _ } => BareCommand::Clear,
+                Command::Save(SaveCmdArgs { dst, .. }) => BareCommand::Save { dst },
+                Command::Quit => BareCommand::Quit,
             }
-            CommandKind::Save => {
-                let Some(dst) = self.inner.args.next() else {
-                    return Ok(Command::Save { dst: "history.nbt" });
-                };
-
-                if self.inner.args.next().is_some() {
-                    return Err(self.err_through_end_from(
-                        2,
-                        "`save` requires exactly one argument, the destination",
-                    ));
-                }
+        }
+    }
 
-                Command::Save { dst }
-            }
-        };
+    fn new_runner() -> CommandRunner<()> {
+        CommandRunner::new_all_disabled()
+            .enable_print_markup(|_| {})
+            .enable_clear(|_| CommandControlFlow::Continue)
+            .enable_save(SessionHistory::new())
+            .enable_quit()
+    }
 
-        Ok(command)
+    fn parser(input: &'static str) -> Option<CommandParser<'static>> {
+        CommandParser::new(input, 0)
     }
-}
 
-#[cfg(test)]
-mod test {
-    use super::*;
+    fn parser_uw(input: &'static str) -> CommandParser<'static> {
+        parser(input).unwrap()
+    }
```

Comment:
This function is only used with `.word_boundaries`. Should it be modified to return them directly?

Minor: You can annotate functions like this with `#[track_caller]` so that a possible unwrap panic will be reported at the call site (where the actual mistake is), not here.